Sources/Swift/Integrations/SessionReplay/SentryTouchTracker.swift

```diff
@@ -0,0 +1,149 @@
+import Foundation
+#if (os(iOS) || os(tvOS)) && !SENTRY_NO_UIKIT
+import UIKit
+
+@objcMembers
+class SentryTouchTracker: NSObject {
+    
+    private struct TouchEvent {
+        let x: CGFloat
+        let y: CGFloat
+        let timestamp: TimeInterval
+        let phase: TouchEventPhase
+        
+        var point: CGPoint {
+            CGPoint(x: x, y: y)
+        }
+    }
+    
+    private class TouchInfo {
+        let id: Int
+        
+        var start: TouchEvent?
+        var end: TouchEvent?
+        var movements = [TouchEvent]()
+        
+        init(id: Int) {
+            self.id = id
+        }
+    }
+    
+    private var trackedTouches = [UITouch: TouchInfo]()
+    private var touchId = 1
+    private let dateProvider: SentryCurrentDateProvider
+    private let scale: CGAffineTransform
+    
+    init(dateProvider: SentryCurrentDateProvider, scale: Float) {
+        self.dateProvider = dateProvider
+        self.scale = CGAffineTransform(scaleX: CGFloat(scale), y: CGFloat(scale))
+    }
+    
+    func trackTouchFrom(event: UIEvent) {
+        guard let touches = event.allTouches else { return }
+        for touch in touches {
+            guard touch.phase == .began || touch.phase == .ended || touch.phase == .moved || touch.phase == .cancelled else { continue }
+            let info = trackedTouches[touch] ?? TouchInfo(id: touchId++)
+            let position = touch.location(in: nil).applying(scale)
+            let newEvent = TouchEvent(x: position.x, y: position.y, timestamp: event.timestamp, phase: touch.phase.toRRWebTouchPhase())
+            
+            switch touch.phase {
+            case .began:
+                info.start = newEvent
+            case .ended, .cancelled:
+                info.end = newEvent
+            case .moved:
+                if let last = info.movements.last, touchesDelta(last.point, position) < 10 { continue }
+                info.movements.append(newEvent)
+                debounceEvents(in: info)
+            default:
+                continue
+            }
+            
+            trackedTouches[touch] = info
+        }
+    }
+    
+    private func touchesDelta(_ lastTouch: CGPoint, _ newTouch: CGPoint) -> CGFloat {
+        let dx = newTouch.x - lastTouch.x
+        let dy = newTouch.y - lastTouch.y
+        return sqrt(dx * dx + dy * dy)
+    }
+    
+    private func debounceEvents(in touchInfo: TouchInfo) {
+        guard touchInfo.movements.count >= 3 else { return }
+        let subset = touchInfo.movements.suffix(3)
+        if subset[subset.startIndex + 2].timestamp - subset[subset.startIndex + 1].timestamp > 0.5 {
```

Comment:
`m`: Shouldn't we compare the last and the first item?
```suggestion
        if subset[subset.startIndex + 2].timestamp - subset[subset.startIndex].timestamp > 0.5 {
```

Reply:
I dont see why, it only matter to compare the time gap between the last two events.